### DISCUSSION TITLE
Como testar mocks de dados de APIs do HTML5 em componentes React, usando o Jest

### DISCUSSION THREAD

**Discussion by felquis** [frontendbr/forum#392] [Perguntas e Respostas]
Fala pessoal, estou trabalhando nos testes do repositório [react-event-components](https://github.com/pagarme/react-event-components/pull/38), usando o react-scripts ejetado com o jest e, para conseguir aumentar o test coverage preciso simular alguns eventos como geolocalização, window resize, touchstart, touchmove, touchend e eu simplesmente travei.

Alguém da comunidade teria uma sugestão de como passar dessa faze? Obrigado :)

**Comment by fdaciuk** [comment]
Não faz mais sentido testar isso usando uma ferramenta de E2E? Recomendo o [Dalek](http://dalekjs.com/) =)

**Comment by jhonnymoreira** [comment]
Na pegada que o @fdaciuk postou: https://github.com/segmentio/nightmare

**Comment by thiamsantos** [comment]
@felquis dá pra usar também a [API de eventos](https://developer.mozilla.org/en-US/docs/Web/API/Event).

```js
const node = document.createElement('div')
node.addEventListener('click', e => {
  console.log(e)
})

const event = document.createEvent('Event')
event.initEvent('click', true, true)

node.dispatchEvent(event)
```
